### ISSUE TITLE
Regression: used-before-assignment error not reported for function arguments

### ISSUE THREAD

**Issue by pbasista** [PyCQA/pylint#2374]
### Steps to reproduce
Run Pylint on the following code:
```python
"""A module to demonstrate Python's UnboundLocalError exception."""

def test(tmp=0):
    """Demonstrate UnboundLocalError exception on a function argument."""
    print(tmp)
    def inner():
        """Python considers "tmp" as a local variable in this function."""
        print(tmp)  # UnboundLocalError is raised here
        tmp = 1
    inner()
    print(tmp)


test()
```
### Current behavior:
No warning or error is reported.

However, the code fails to run because of [this](https://docs.python.org/3/faq/programming.html#why-am-i-getting-an-unboundlocalerror-when-the-variable-has-a-value) well-known issue with using variables considered as local before their assignment. The following exception is raised:
```
UnboundLocalError: local variable 'tmp' referenced before assignment
```
### Expected behavior
`used-before-assignment` error is reported on line where `tmp` is first used in function `inner`.

This error is correctly reported by Pylint 1.9.3, but not by Pylint versions 2 and above.

A similar error is correctly reported by Pylint versions 2 and above when the `tmp` variable is locally defined within the outer function, but not when it is a function's argument.

### pylint --version output
pylint 2.1.0
astroid 2.0.2
Python 3.6.6 (default, Jun 27 2018, 14:44:17) 
[GCC 8.1.0]
(the current master branch version 7be00a961bbfebcdd7a9b955f974c30c5254a64e)

**Comment by PCManticore** [comment]
Thanks for the report, I can also reproduce this regression.
